src/filters/fb_network.rs

```diff
@@ -0,0 +1,307 @@
+                .map(|s| self.builder.create_string(s))
+                .collect();
+            Some(self.builder.create_vector(&offsets))
+        } else {
+            None
+        };
+
+        let raw_line = network_filter
+            .raw_line
+            .as_ref()
+            .map(|v| self.builder.create_string(v.as_str()));
+
+        let filter = fb::NetworkFilter::create(
+            &mut self.builder,
+            &fb::NetworkFilterArgs {
+                mask: network_filter.mask.bits(),
+                patterns: patterns,
+                modifier_option: modifier_option,
+                opt_domains: opt_domains,
+                opt_not_domains: opt_not_domains,
+                hostname: hostname,
+                tag: tag,
+                raw_line: raw_line,
+            },
+        );
+
+        self.filters.push(filter);
+        u32::try_from(self.filters.len() - 1).expect("< u32::MAX")
+    }
+
+    pub fn finish(&mut self) -> Vec<u8> {
+        let filters = self.builder.create_vector(&self.filters);
+
+        let unique_domains = self.builder.create_vector(&self.unique_domains);
+
+        let storage = fb::NetworkFilterList::create(
+            &mut self.builder,
+            &&fb::NetworkFilterListArgs {
+                network_filters: Some(filters),
+                unique_domains_hashes: Some(unique_domains),
+            },
+        );
+        self.builder.finish(storage, None);
+
+        let binary = Vec::from(self.builder.finished_data());
+        binary
+    }
+}
+pub struct FlatPatterns<'a> {
+    patterns: Option<flatbuffers::Vector<'a, flatbuffers::ForwardsUOffset<&'a str>>>,
+}
+
+impl<'a> FlatPatterns<'a> {
+    #[inline(always)]
+    pub fn new(
+        patterns: Option<flatbuffers::Vector<'a, flatbuffers::ForwardsUOffset<&'a str>>>,
+    ) -> Self {
+        Self { patterns }
+    }
+
+    #[inline(always)]
+    pub fn iter(&self) -> FlatPatternsIterator {
+        FlatPatternsIterator {
+            patterns: self,
+            len: self.patterns.map_or(0, |d| d.len()),
+            index: 0,
+        }
+    }
+}
+
+pub struct FlatPatternsIterator<'a> {
+    patterns: &'a FlatPatterns<'a>,
+    len: usize,
+    index: usize,
+}
+
+impl<'a> Iterator for FlatPatternsIterator<'a> {
+    type Item = &'a str;
+
+    #[inline(always)]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.patterns.patterns.map_or(None, |fi| {
+            if self.index < self.len {
+                self.index += 1;
+                Some(fi.get(self.index - 1))
+            } else {
+                None
+            }
+        })
+    }
+}
+
+// Implement ExactSizeIterator for FilterPartIterator
+impl<'a> ExactSizeIterator for FlatPatternsIterator<'a> {
+    #[inline(always)]
+    fn len(&self) -> usize {
+        self.len
+    }
+}
+
+pub struct FlatNetworkFilter<'a> {
+    key: u64,
+    owner: &'a FlatNetworkFilterList,
+    fb_filter: &'a fb::NetworkFilter<'a>,
+
+    pub mask: NetworkFilterMask,
```

Comment:
it's not clear why `mask` is pub here, but the other fields are not.

Reply:
because it is used outside, in the `FlatNetworkFilterList::check`

Reply:
should it be `pub(crate)`?

Reply:
it could. done